force-app/main/default/classes/RecurrenceRule.cls

```diff
@@ -100,11 +100,85 @@ public with sharing class RecurrenceRule {
 
     public String build() {
         // TODO: build the rule string from the RecurrenceRule object
+        // FREQ (required)
+        String rRule = 'FREQ=' + this.freq;
+
+        // INTERVAL
+        if (this.interval != 1) {
+            rRule += ';INTERVAL=' + this.interval;
+        }
+
+        // UNTIL
+        if (this.endDate != null) {
+            String endDateString = Datetime.newInstanceGMT(
+                    endDate,
+                    Time.newInstance(0, 0, 0, 0)
+                )
+                .formatGmt('yyyyMMdd');
+
+            rRule += ';UNTIL=' + endDateString + 'T000000Z';
```

Comment:
Still learning this RRule string but had a question here, should 'T000000Z' be hardcoded while building the string